layer/common_logging.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <bits/types/FILE.h>
+
+#include <cassert>
+
+#include "event_logging.h"
+#include "layer_data.h"
+
+namespace performancelayers {
+// Converts `event` to a string with the common log format. The common log
+// format looks like this:
+// ```c++
+// event_name,attribute1_name:attribute1_value,attribute2_name:attribute2_value,...
+// ```
+std::string EventToCommonLogStr(Event &event);
+
+// CommonLogger logs the events in the common log. Sample use:
+// ```c++
+// CommonLogger logger("event.log");
+// logger.StartLog();
+// Event compile_time_event = ...;
+// logger.AddEvent(&compile_time_event);
+// logger.Flush();
+// logger.EndLog();
```

Comment:
This sample use applies to all loggers, could we move it to the documentation of `EventLogger`?